src/test/kotlin/uk/gov/justice/digital/hmpps/incidentreporting/service/NomisSyncServiceTest.kt

```diff
@@ -285,14 +290,17 @@ class NomisSyncServiceTest {
       initialMigration = initialMigration,
     )
 
-    whenever(reportRepository.save(any())).thenReturn(sampleReport)
```

Comment:
having the repository return `sampleReport` meant that calling `assertSampleReportConvertedToDto` further down did not in fact test the result of the upsert!

instead, the mocked save method now returns the real newly-created report.